sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -984,6 +993,20 @@ public partial class HttpPipelineTransportOptions
         public System.Collections.Generic.IList<System.Security.Cryptography.X509Certificates.X509Certificate2> ClientCertificates { get { throw null; } }
         public System.Func<Azure.Core.Pipeline.ServerCertificateCustomValidationArgs, bool>? ServerCertificateCustomValidationCallback { get { throw null; } set { } }
     }
+    public abstract partial class RetryPolicy : Azure.Core.Pipeline.HttpPipelinePolicy
+    {
+        protected RetryPolicy(Azure.Core.RetryOptions? options = null) { }
+        protected virtual System.TimeSpan CalculateNextDelay(Azure.Core.HttpMessage message) { throw null; }
+        protected virtual System.Threading.Tasks.ValueTask<System.TimeSpan> CalculateNextDelayAsync(Azure.Core.HttpMessage message) { throw null; }
+        protected virtual void OnReceivedResponse(Azure.Core.HttpMessage message) { }
+        protected virtual System.Threading.Tasks.ValueTask OnReceivedResponseAsync(Azure.Core.HttpMessage message) { throw null; }
+        protected virtual void OnSendingRequest(Azure.Core.HttpMessage message) { }
+        protected virtual System.Threading.Tasks.ValueTask OnSendingRequestAsync(Azure.Core.HttpMessage message) { throw null; }
+        public override void Process(Azure.Core.HttpMessage message, System.ReadOnlyMemory<Azure.Core.Pipeline.HttpPipelinePolicy> pipeline) { }
```

Comment:
I don't think so because user's may want to inject logic into Process but still call the base method, as in the GlobalTimeoutPolicy scenario:

```c#
internal class GlobalTimeoutPolicy : RetryPolicy
{
    private readonly TimeSpan _timeout;
    public GlobalTimeoutPolicy(TimeSpan timeout)
    {
        _timeout = timeout;
    }

    public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
    {
        ProcessAsyncInternal(message, pipeline, false).EnsureCompleted();
    }

    public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
    {
        await ProcessAsyncInternal(message, pipeline, true).ConfigureAwait(false);
    }

    private async ValueTask ProcessAsyncInternal(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline, bool async)
    {
        // create a linked token source that we will use to cancel pipeline activity if the timeout is reached
        using var cts = CancellationTokenSource.CreateLinkedTokenSource(message.CancellationToken);
        cts.CancelAfter(_timeout);

        // replace the original cancellation token with the linked one, but store off the original
        var originalToken = message.CancellationToken;
        message.CancellationToken = cts.Token;

        try
        {
            if (async)
            {
                await base.ProcessAsync(message, pipeline).ConfigureAwait(false);
            }
            else
            {
                base.Process(message, pipeline);
            }
        }
        finally
        {
            // reset the original cancellation token
            message.CancellationToken = originalToken;
        }
    }
}
```

Reply:
OK. Makes sense.